From 8416d8a1a423535fbc5d4e7416d6eac8ac5c050b Mon Sep 17 00:00:00 2001
From: Preston Maness <aggroskater@gmail.com>
Date: Tue, 26 Apr 2016 23:06:57 -0500
Subject: [PATCH] Fix #1530 (null pointer dereference)

---
 client/hostinfo_unix.cpp | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/client/hostinfo_unix.cpp b/client/hostinfo_unix.cpp
index 38a0cf5..54f577e 100644
--- a/client/hostinfo_unix.cpp
+++ b/client/hostinfo_unix.cpp
@@ -1989,11 +1989,12 @@ const vector<string> X_display_values_initialize() {
         "[idle_detection] Error (%d) opening %s.", errno, dir.c_str());
     }
   }
-
-  while ((dirp = readdir(dp)) != NULL) {
-    display_values.push_back(string(dirp->d_name));
+  else {
+    while ((dirp = readdir(dp)) != NULL) {
+      display_values.push_back(string(dirp->d_name));
+    }
+    closedir(dp);
   }
-  closedir(dp);
 
   // Get rid of non-matching elements and format the matching ones.
   for ( it = display_values.begin() ; it != display_values.end() ; ) {
