Index: boinc_debian_new/html/Makefile.am
===================================================================
--- boinc_debian_new.orig/html/Makefile.am
+++ boinc_debian_new/html/Makefile.am
@@ -1,4 +1,4 @@
 htmldir=$(DESTDIR)/$(prefix)/share/boinc-server-maker/html
 install-data-local:
 	mkdir -p $(htmldir)
-	cp -r inc languages ops user $(htmldir)
+	cp -r bt drupal inc languages ops project.sample queue user $(htmldir)
