Index: boinc_debian_new/Makefile.am
===================================================================
--- boinc_debian_new.orig/Makefile.am
+++ boinc_debian_new/Makefile.am
@@ -16,7 +16,7 @@
 endif
 
 if ENABLE_SERVER
-   SERVER_SUBDIRS = db test py sched apps tools vda html
+   SERVER_SUBDIRS = db test py sched apps tools vda html samples
 ## once contained samples/example_app which breaks out-of-source-tree builds
 endif
 
@@ -33,7 +33,7 @@
 
 # ORDER MATTERS below.  One must build dependencies FIRST, then things
 # that depend on them later
-SUBDIRS = m4 $(API_SUBDIRS) $(SERVER_SUBDIRS) $(CLIENT_SUBDIRS) $(CLIENTGUI_SUBDIRS) doc
+SUBDIRS = m4 $(API_SUBDIRS) $(CLIENT_SUBDIRS) $(CLIENTGUI_SUBDIRS) $(SERVER_SUBDIRS) doc
 
 # Putting a directory name recursively copies the entire contents - the
 # dist-hook below gets rid of CVS directories.
Index: boinc_debian_new/samples/Makefile.am
===================================================================
--- /dev/null
+++ boinc_debian_new/samples/Makefile.am
@@ -0,0 +1 @@
+SUBDIRS=wrapper
