Index: boinc/lib/Makefile.am
===================================================================
--- boinc.orig/lib/Makefile.am
+++ boinc/lib/Makefile.am
@@ -253,7 +253,7 @@ msg_test_CXXFLAGS = $(PTHREAD_CFLAGS)
 msg_test_LDADD = $(LIBBOINC)
 crypt_prog_SOURCES = crypt_prog.cpp 
 crypt_prog_CXXFLAGS = $(PTHREAD_CFLAGS) $(SSL_CXXFLAGS)
-crypt_prog_LDADD = $(LIBBOINC_CRYPT_STATIC) $(LIBBOINC) $(SSL_LIBS) 
+crypt_prog_LDADD = $(LIBBOINC_CRYPT) $(LIBBOINC) $(SSL_LIBS)
 parse_test_SOURCES = parse_test.cpp 
 parse_test_CXXFLAGS = $(PTHREAD_CFLAGS)
 parse_test_LDADD = $(LIBBOINC)
Index: boinc/samples/example_app/Makefile
===================================================================
--- boinc.orig/samples/example_app/Makefile
+++ boinc/samples/example_app/Makefile
@@ -60,29 +60,32 @@ install: uc2
 # specify library paths explicitly (rather than -l)
 # because otherwise you might get a version in /usr/lib etc.
 
+.cpp.o:
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -c -fPIE $<
+
 uc2: uc2.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2 uc2.o libstdc++.a -pthread \
-	$(BOINC_API_DIR)/libboinc_api.a \
-	$(BOINC_LIB_DIR)/libboinc.a
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2 uc2.o -lstdc++ -pthread \
+	$(BOINC_API_DIR)/.libs/libboinc_api.so \
+	$(BOINC_LIB_DIR)/.libs/libboinc.so
 
 ucn: ucn.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o ucn ucn.o libstdc++.a -pthread \
-	$(BOINC_API_DIR)/libboinc_api.a \
-	$(BOINC_LIB_DIR)/libboinc.a
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o ucn ucn.o -lstdc++ -pthread \
+	$(BOINC_API_DIR)/.libs/libboinc_api.so \
+	$(BOINC_LIB_DIR)/.libs/libboinc.so
 
 uc2_graphics: uc2_graphics.o ttfont.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_graphics2.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2_graphics uc2_graphics.o ttfont.o libstdc++.a -pthread \
-	$(BOINC_API_DIR)/libboinc_graphics2.a \
-	$(BOINC_API_DIR)/libboinc_api.a \
-	$(BOINC_LIB_DIR)/libboinc.a \
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2_graphics uc2_graphics.o ttfont.o -lstdc++ -pthread \
+	$(BOINC_API_DIR)/.libs/libboinc_graphics2.so \
+	$(BOINC_API_DIR)/.libs/libboinc_api.so \
+	$(BOINC_LIB_DIR)/.libs/libboinc.so \
     $(LIBGLUT) $(LIBGLU) $(LIBJPEG) \
     -lftgl -lGL -lX11 -lXmu -lm
 
 slide_show: slide_show.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_graphics2.a
-	$(CXX) $(CXXFLAGS) -o slide_show slide_show.o libstdc++.a -pthread \
-	$(BOINC_API_DIR)/libboinc_graphics2.a \
-	$(BOINC_API_DIR)/libboinc_api.a \
-	$(BOINC_LIB_DIR)/libboinc.a \
-	$(BOINC_ZIP_DIR)/libboinc_zip.a \
+	$(CXX) $(CXXFLAGS) -o slide_show slide_show.o -lstdc++ -pthread \
+	$(BOINC_API_DIR)/.libs/libboinc_graphics2.so \
+	$(BOINC_API_DIR)/.libs/libboinc_api.so \
+	$(BOINC_LIB_DIR)/.libs/libboinc.so \
+	$(BOINC_ZIP_DIR)/.libs/libboinc_zip.so \
     $(LIBGLUT) $(LIBGLU) $(LIBJPEG) \
     -lGL -lX11 -lXmu -lm
Index: boinc/samples/sleeper/Makefile
===================================================================
--- boinc.orig/samples/sleeper/Makefile
+++ boinc/samples/sleeper/Makefile
@@ -16,13 +16,16 @@ PROGS = sleeper
 
 all: $(PROGS)
 
-libstdc++.a:
-	ln -s `g++ -print-file-name=libstdc++.a`
+#libstdc++.a:
+#	ln -s `g++ -print-file-name=libstdc++.a`
 
 clean: distclean
 
 distclean:
 	rm -f $(PROGS) *.o libstdc++.a
 
-sleeper: sleeper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
-	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o sleeper sleeper.o libstdc++.a -pthread -lboinc_api -lboinc
+.cpp.o:
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -fPIE -c $<
+
+sleeper: sleeper.o $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o sleeper sleeper.o -lstdc++ -pthread $(BOINC_API_DIR)/.libs/libboinc_api.so $(BOINC_LIB_DIR)/.libs/libboinc.so
Index: boinc/samples/wrapper/Makefile
===================================================================
--- boinc.orig/samples/wrapper/Makefile
+++ boinc/samples/wrapper/Makefile
@@ -37,5 +37,13 @@ REGEXP_OBJS = \
 	regexp_memory.o \
 	regexp_report.o
 
-wrapper: wrapper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(REGEXP_OBJS)
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) libstdc++.a $(PTHREAD) -lboinc_api -lboinc -lboinc_zip $(LDFLAGS)
+.cpp.o:
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -c -fPIC $<
+.c.o:
+	$(CC) $(CFLAGS) $(CPPFLAGS) -c -fPIC $<
+
+wrapper: wrapper.o $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(REGEXP_OBJS)
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) -lstdc++ $(PTHREAD) \
+		$(BOINC_API_DIR)/.libs/libboinc_api.so \
+		$(BOINC_LIB_DIR)/.libs/libboinc.so \
+		$(BOINC_ZIP_DIR)/.libs/libboinc_zip.so $(LDFLAGS)
Index: boinc/samples/worker/Makefile
===================================================================
--- boinc.orig/samples/worker/Makefile
+++ boinc/samples/worker/Makefile
@@ -9,5 +9,8 @@ clean: distclean
 distclean:
 	rm -f $(PROGS) *.o
 
+.cpp.o:
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -c -fPIC $<
+
 worker: worker.o
-	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o worker worker.o
+	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o worker worker.o
