--- a/lib/boinc_win.h
+++ b/lib/boinc_win.h
@@ -197,7 +197,7 @@
 #include <assert.h>
 #include <ctype.h>
 #include <errno.h>
-#include <math.h>
+#include <cmath>
 #include <setjmp.h>
 #include <signal.h>
 #include <stdarg.h>
--- a/lib/md5.c
+++ b/lib/md5.c
@@ -88,7 +88,7 @@
  * For reference, here is the program that computed the T values.
  */
 #if 0
-#include <math.h>
+#include <cmath>
 main()
 {
     int i;
