Author: Steffen Moeller <moeller@debian.org>
Description: Initialisation.

Index: boinc/lib/crypt.cpp
===================================================================
--- boinc.orig/lib/crypt.cpp
+++ boinc/lib/crypt.cpp
@@ -342,6 +342,7 @@
     char clear_buf[MD5_LEN];
     int n, retval;
     DATA_BLOCK clear_signature;
+    clear_buf[0]=0;
 
     n = (int)strlen(md5_buf);
     clear_signature.data = (unsigned char*)clear_buf;
