--- boinc-7.4.14+dfsg.orig/Makefile.incl
+++ boinc-7.4.14+dfsg/Makefile.incl
@@ -43,22 +43,22 @@ AM_LDFLAGS =
 
 LIBSCHED = $(top_builddir)/sched/libsched.la
 $(LIBSCHED):
-	cd $(top_builddir)/sched; ${MAKE} libsched.la
+	cd $(top_builddir)/sched; ${MAKE} libsched.a
 LIBSCHED_FCGI = $(top_builddir)/sched/libsched_fcgi.la
 $(LIBSCHED_FCGI):
-	cd $(top_builddir)/sched; ${MAKE} libsched_fcgi.la
+	cd $(top_builddir)/sched; ${MAKE} libsched_fcgi.a
 LIBBOINC = $(top_builddir)/lib/libboinc.la
 $(LIBBOINC):
-	cd $(top_builddir)/lib; ${MAKE} libboinc.la
+	cd $(top_builddir)/lib; ${MAKE} libboinc.a
 LIBBOINC_CRYPT = $(top_builddir)/lib/libboinc_crypt.la
 $(LIBBOINC_CRYPT):
-	cd $(top_builddir)/lib; ${MAKE} libboinc_crypt.la
+	cd $(top_builddir)/lib; ${MAKE} libboinc_crypt.a
 LIBBOINC_FCGI = $(top_builddir)/lib/libboinc_fcgi.la
 $(LIBBOINC_FCGI):
-	cd $(top_builddir)/lib; ${MAKE} libboinc_fcgi.la
+	cd $(top_builddir)/lib; ${MAKE} libboinc_fcgi.a
 LIBAPI = $(top_builddir)/api/libboinc_api.la
 $(LIBAPI):
-	cd $(top_builddir)/api; ${MAKE} libboinc_api.la
+	cd $(top_builddir)/api; ${MAKE} libboinc_api.a
 
 SERVERLIBS = $(LIBSCHED) $(LIBBOINC_CRYPT) $(LIBBOINC) $(MYSQL_LIBS) $(PTHREAD_LIBS) $(RSA_LIBS) $(SSL_LIBS)
 SERVERLIBS_FCGI = $(LIBSCHED_FCGI) $(LIBBOINC_CRYPT) $(LIBBOINC_FCGI) -lfcgi $(MYSQL_LIBS) $(PTHREAD_LIBS) $(RSA_LIBS) $(SSL_LIBS)
