Author: Jussi Lahtinen
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1274456

Index: boinc/client/hostinfo_unix.cpp
===================================================================
--- boinc.orig/client/hostinfo_unix.cpp
+++ boinc/client/hostinfo_unix.cpp
@@ -1890,6 +1890,7 @@
     static Display* disp = NULL;
     
     long idle_time = 0;
+    int  event_base_return, error_base_return;
     
     if(disp != NULL) {
         XScreenSaverQueryInfo(disp, DefaultRootWindow(disp), xssInfo);
@@ -1946,6 +1947,9 @@
         // or DISPLAY points to wrong/invalid display
         if(disp != NULL) {
             xssInfo = XScreenSaverAllocInfo();
+            if (!XScreenSaverQueryExtension(disp, &event_base_return, &error_base_return)){
+               // How to handle failing?
+            }
         }
     }
 
