Description: Fix localization missing and install path
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Author: Josef Andersson (northar)

--- boinc-7.6.11+dfsg.orig/locale/Makefile.am
+++ boinc-7.6.11+dfsg/locale/Makefile.am
@@ -16,16 +16,20 @@ locale_dirs = \
 	el \
 	es \
 	fa \
+	fa_IR \
 	fi \
 	fr \
 	gl \
+	he \
 	hr \
 	hu \
 	it_IT \
 	ja \
+	ka \
 	ko \
 	lt \
 	lv \
+	ms \
 	nb \
 	nl \
 	nn \
@@ -36,7 +40,7 @@ locale_dirs = \
 	ru \
 	sk \
 	sl \
-	sv_SE \
+	sv \
 	tr \
 	uk \
 	zh_CN \
@@ -45,12 +49,12 @@ locale_dirs = \
 install-data-hook:
 	mydir=$(DESTDIR)$(datadir) ;\
 	for ldir in $(locale_dirs) ; do \
-	  $(INSTALL) -d $$mydir/$$ldir ; \
+	  $(INSTALL) -d $$mydir/$$ldir/LC_MESSAGES ; \
 	  if [ -f $$ldir/BOINC-Manager.mo ] ; then \
-	    $(INSTALL_DATA) $$ldir/BOINC-Manager.mo $$mydir/$$ldir/BOINC-Manager.mo ; \
+	    $(INSTALL_DATA) $$ldir/BOINC-Manager.mo $$mydir/$$ldir/LC_MESSAGES/BOINC-Manager.mo ; \
 	  fi ;\
 	  if [ -f $$ldir/BOINC-Client.mo ] ; then \
-	    $(INSTALL_DATA) $$ldir/BOINC-Client.mo $$mydir/$$ldir/BOINC-Client.mo ; \
+	    $(INSTALL_DATA) $$ldir/BOINC-Client.mo $$mydir/$$ldir/LC_MESSAGES/BOINC-Client.mo ; \
 	  fi ;\
 	done
 
