Description: defining PATH_MAX to fix hurd build
Author: Gianfranco Costamagna <locutusofborg@debian.org>
Last-Update: 2017-09-17

Index: boinc-v2/clientgui/BOINCGUIApp.cpp
===================================================================
--- boinc-v2.orig/clientgui/BOINCGUIApp.cpp
+++ boinc-v2/clientgui/BOINCGUIApp.cpp
@@ -49,6 +49,9 @@
 #include "procinfo.h"
 #include "sg_BoincSimpleFrame.h"
 
+#ifndef PATH_MAX
+#define PATH_MAX 4096
+#endif
 
 bool s_bSkipExitConfirmation = false;
 
