--- a/api/boinc_api.cpp
+++ b/api/boinc_api.cpp
@@ -353,7 +353,7 @@
 
     if (standalone) return true;
 
-    sprintf(msg_buf,
+    snprintf(msg_buf,sizeof(msg_buf),
         "<current_cpu_time>%e</current_cpu_time>\n"
         "<checkpoint_cpu_time>%e</checkpoint_cpu_time>\n",
         cpu_t, cp_cpu_t
@@ -632,6 +632,10 @@
 }
 
 int boinc_get_status(BOINC_STATUS *s) {
+    if (!s) {
+        fprintf(stderr,"E: boinc_get_status(NULL)\n");
+	return 1;
+    }
     s->no_heartbeat = boinc_status.no_heartbeat;
     s->suspended = boinc_status.suspended;
     s->quit_request = boinc_status.quit_request;
@@ -808,7 +812,7 @@
 int boinc_parse_init_data_file() {
     FILE* f;
     int retval;
-    char buf[256];
+    char buf[2560];
 
     if (aid.project_preferences) {
         free(aid.project_preferences);
@@ -858,15 +862,15 @@
     );
     if (other_pid) {
         sprintf(buf, "<other_pid>%d</other_pid>\n", other_pid);
-        strcat(msg_buf, buf);
+        strlcat(msg_buf, buf, sizeof(msg_buf));
     }
     if (_bytes_sent) {
         sprintf(buf, "<bytes_sent>%f</bytes_sent>\n", _bytes_sent);
-        strcat(msg_buf, buf);
+        strlcat(msg_buf, buf, sizeof(msg_buf));
     }
     if (_bytes_received) {
         sprintf(buf, "<bytes_received>%f</bytes_received>\n", _bytes_received);
-        strcat(msg_buf, buf);
+        strlcat(msg_buf, buf, sizeof(msg_buf));
     }
     if (app_client_shm->shm->app_status.send_msg(msg_buf)) {
         return 0;
@@ -949,7 +953,7 @@
 }
 
 static void handle_upload_file_status() {
-    char path[MAXPATHLEN], buf[256], log_name[256], *p, log_buf[256];
+    char path[MAXPATHLEN], buf[MAXPATHLEN], log_name[MAXPATHLEN], *p, log_buf[256];
     std::string filename;
     int status;
 
@@ -1281,7 +1285,7 @@
     // send graphics-related messages
     //
     if (send_web_graphics_url && !app_client_shm->shm->graphics_reply.has_msg()) {
-        sprintf(buf,
+        snprintf(buf, sizeof(buf),
             "<web_graphics_url>%s</web_graphics_url>",
             web_graphics_url
         );
@@ -1289,7 +1293,7 @@
         send_web_graphics_url = false;
     }
     if (send_remote_desktop_addr && !app_client_shm->shm->graphics_reply.has_msg()) {
-        sprintf(buf,
+        snprintf(buf, sizeof(buf),
             "<remote_desktop_addr>%s</remote_desktop_addr>",
             remote_desktop_addr
         );
@@ -1505,7 +1509,7 @@
         DirScanner dirscan(path);
         while (dirscan.scan(filename)) {
             if (strstr(filename.c_str(), "trickle_down")) {
-                strncpy(buf, filename.c_str(), len);
+                strlcpy(buf, filename.c_str(), len);
                 return true;
             }
         }
