Index: boinc-7.16.1+dfsg/clientgui/AdvancedFrame.cpp
===================================================================
--- boinc-7.16.1+dfsg.orig/clientgui/AdvancedFrame.cpp
+++ boinc-7.16.1+dfsg/clientgui/AdvancedFrame.cpp
@@ -344,11 +344,13 @@
         _("Select computer...\tCtrl+Shift+I"),
         _("Connect to a BOINC client on another computer")
     );
+#ifndef __WXGTK__
     menuFile->Append(
         ID_SHUTDOWNCORECLIENT, 
         _("Shut down connected client..."),
         _("Shut down the currently connected BOINC client")
     );
+#endif
     menuFile->AppendSeparator();
 
     strMenuDescription.Printf(
