Index: boinc-7.16.1+dfsg/clientgui/AdvancedFrame.cpp
===================================================================
--- boinc-7.16.1+dfsg.orig/clientgui/AdvancedFrame.cpp
+++ boinc-7.16.1+dfsg/clientgui/AdvancedFrame.cpp
@@ -363,6 +363,18 @@
         strMenuDescription
     );
 
+/* 
+   On Linux distributions, BOINC runs as a service. Users must not be able to
+   try stopping the service from exit menu entry.
+   This leads to unexpected behaviour, like:
+   - service being killed;
+   - service still running.
+   Moreover, the Manager will no longer be able to connect to the client, unless
+   the user connects to 127.0.0.1. Then if the Manager is connected to the client
+   by using 127.0.0.1 address, the "Exit from BOINC Manager" entry will not
+   show any frame asking the user if he wants to stop the service.
+*/
+#ifndef __WXGTK__
     strMenuDescription.Printf(
         _("Exit %s"), 
         pSkinAdvanced->GetApplicationName().c_str()
@@ -383,6 +395,7 @@
         strMenuName,
         strMenuDescription
     );
+#endif
 
 #ifdef __WXMAC__
     // wxWidgets actually puts this in the BOINCManager menu
Index: boinc-7.16.1+dfsg/clientgui/sg_BoincSimpleFrame.cpp
===================================================================
--- boinc-7.16.1+dfsg.orig/clientgui/sg_BoincSimpleFrame.cpp
+++ boinc-7.16.1+dfsg/clientgui/sg_BoincSimpleFrame.cpp
@@ -154,6 +154,18 @@
         strMenuDescription
     );
 
+/* 
+   On Linux distributions, BOINC runs as a service. Users must not be able to
+   try stopping the service from exit menu entry.
+   This leads to unexpected behaviour, like:
+   - service being killed;
+   - service still running.
+   Moreover, the Manager will no longer be able to connect to the client, unless
+   the user connects to 127.0.0.1. Then if the Manager is connected to the client
+   by using 127.0.0.1 address, the "Exit from BOINC Manager" entry will not
+   show any frame asking the user if he wants to stop the service.
+*/
+#ifndef __WXGTK_
     strMenuDescription.Printf(
         _("Exit %s"),
         pSkinAdvanced->GetApplicationName().c_str()
@@ -169,6 +181,7 @@
         strMenuName,
         strMenuDescription
     );
+#endif
 
 #ifdef __WXMAC__
     // wxWidgets actually puts this in the BOINCManager menu
