Author: Steffen Moeller <moeller@debian.org>
Description: Improved memory handling.

--- a/lib/procinfo_win.cpp
+++ b/lib/procinfo_win.cpp
@@ -25,6 +25,9 @@
     tNTQSI   pNTQSI = (tNTQSI)GetProcAddress(hNTDllLib, "NtQuerySystemInformation");
     ULONG    cbBuffer = 0;
 
+    if (!ppBuffer) return ERR_NULL;
+    if (!pcbBuffer) return ERR_NULL;
+
     while (1) {
         // Store the buffer size since it appears that somebody is monkeying around
         //   with the return values on some systems.
