Index: boinc/lib/Makefile.am
===================================================================
--- boinc.orig/lib/Makefile.am
+++ boinc/lib/Makefile.am
@@ -176,6 +176,10 @@ lib_LTLIBRARIES = libboinc.la
 libboinc_la_SOURCES = $(generic_sources) $(mac_sources) $(win_sources)
 libboinc_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
+libboinc_la_AR = gcc-ar
+libboinc_la_RANLIB = gcc-ranlib
+libboinc_a_AR = gcc-ar
+libboinc_a_RANLIB = gcc-ranlib
 libboinc_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)
 libboinc_la_LIBADD =
 
@@ -184,6 +188,10 @@ lib_LTLIBRARIES += libboinc_crypt.la
 libboinc_crypt_la_SOURCES = crypt.cpp
 libboinc_crypt_la_CFLAGS = $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CFLAGS)
 libboinc_crypt_la_CXXFLAGS = $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS) $(SSL_CXXFLAGS)
+libboinc_crypt_la_AR = gcc-ar
+libboinc_crypt_la_RANLIB = gcc-ranlib
+libboinc_crypt_a_AR = gcc-ar
+libboinc_crypt_a_RANLIB = gcc-ranlib
 libboinc_crypt_la_LDFLAGS = -version-number $(LIBBOINC_VERSION) $(SSL_LIBS) $(RSA_LIBS)
 libboinc_crypt_la_LIBADD = $(LIBBOINC)
 endif
@@ -193,6 +201,10 @@lib_LTLIBRARIES += libboinc_fcgi.la
 libboinc_fcgi_la_SOURCES = $(libfcgi_sources) $(mac_sources) $(win_sources)
 libboinc_fcgi_la_CFLAGS = -D_USING_FCGI_ $(AM_CFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
 libboinc_fcgi_la_CXXFLAGS = -D_USING_FCGI_ $(AM_CXXFLAGS) $(PICFLAGS) $(PTHREAD_CFLAGS)
+libboinc_fcgi_la_AR = gcc-ar
+libboinc_fcgi_la_RANLIB = gcc-ranlib
+libboinc_fcgi_a_AR = gcc-ar
+libboinc_fcgi_a_RANLIB = gcc-ranlib
 libboinc_fcgi_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)
 libboinc_fcgi_la_LIBADD =
 endif 
@@ -256,7 +268,7 @@ msg_test_CXXFLAGS = $(PTHREAD_CFLAGS)
 msg_test_LDADD = $(LIBBOINC)
 crypt_prog_SOURCES = crypt_prog.cpp 
 crypt_prog_CXXFLAGS = $(PTHREAD_CFLAGS) $(SSL_CXXFLAGS)
-crypt_prog_LDADD = $(LIBBOINC_CRYPT_STATIC) $(LIBBOINC) $(SSL_LIBS) 
+crypt_prog_LDADD = $(LIBBOINC_CRYPT) $(LIBBOINC) $(SSL_LIBS)
 parse_test_SOURCES = parse_test.cpp 
 parse_test_CXXFLAGS = $(PTHREAD_CFLAGS)
 parse_test_LDADD = $(LIBBOINC)
Index: boinc/samples/condor/Makefile
===================================================================
--- boinc.orig/samples/condor/Makefile
+++ boinc/samples/condor/Makefile
@@ -1,14 +1,17 @@
 all: boinc_gahp
 
 clean:
-	rm boinc_gahp
+	rm boinc_gahp libcurl.a
 
-boinc_gahp: boinc_gahp.cpp ../../lib/remote_submit.h ../../lib/remote_submit.cpp
-	g++ -g -O0 -I../../lib \
+libcurl.a:
+	ln -s `$(CXX) -print-file-name=libcurl.a`
+
+boinc_gahp: boinc_gahp.cpp ../../lib/remote_submit.h ../../lib/remote_submit.cpp $(BOINC_LIB_DIR)/libboinc.a
+	$(CXX) $(CPPFLAGS) -static -pthread -g -O0 -I../../lib \
 	-o boinc_gahp boinc_gahp.cpp ../../lib/remote_submit.cpp \
-	-L../../lib -lboinc -lpthread -lcurl
+	$(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a libcurl.a
 
 test: test.cpp ../../lib/remote_submit.cpp ../../lib/remote_submit.h
-	g++ -g -o test -I../../lib \
+	$(CXX) $(CPPFLAGS) -static -g -o test -I../../lib \
 	test.cpp ../../lib/remote_submit.cpp \
-	-L../../lib -lboinc -lcurl
+	$(BOINC_LIB_DIR)/libboinc.a libcurl.a
Index: boinc/samples/multi_thread/Makefile
===================================================================
--- boinc.orig/samples/multi_thread/Makefile
+++ boinc/samples/multi_thread/Makefile
@@ -26,4 +26,4 @@ distclean:
 	/bin/rm -f $(PROGS) *.o libstdc++.a
 
 multi_thread: multi_thread.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
-	$(CXX) $(CXXFLAGS) -o multi_thread multi_thread.o libstdc++.a -pthread -lboinc_api -lboinc
+	$(CXX) $(CPPFLAGS) -static -pthread -o multi_thread multi_thread.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
Index: boinc/samples/sleeper/Makefile
===================================================================
--- boinc.orig/samples/sleeper/Makefile
+++ boinc/samples/sleeper/Makefile
@@ -17,7 +17,7 @@ PROGS = sleeper
 all: $(PROGS)
 
 libstdc++.a:
-	ln -s `g++ -print-file-name=libstdc++.a`
+	ln -s `$(CXX) -print-file-name=libstdc++.a`
 
 clean: distclean
 
@@ -25,4 +25,4 @@ distclean:
 	rm -f $(PROGS) *.o libstdc++.a
 
 sleeper: sleeper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
-	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o sleeper sleeper.o libstdc++.a -pthread -lboinc_api -lboinc
+	$(CXX) $(CPPFLAGS) -static -pthread -o sleeper sleeper.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
Index: boinc/samples/vboxmonitor/Makefile
===================================================================
--- boinc.orig/samples/vboxmonitor/Makefile
+++ boinc/samples/vboxmonitor/Makefile
@@ -27,4 +27,4 @@ distclean:
 	/bin/rm -f $(PROGS) *.o libstdc++.a
 
 vboxmonitor: vboxmonitor.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
-	g++ $(CXXFLAGS) -o vboxmonitor vboxmonitor.o libstdc++.a -pthread -lboinc_api -lboinc
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -static -pthread -o vboxmonitor vboxmonitor.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
Index: boinc/samples/vboxwrapper/Makefile
===================================================================
--- boinc.orig/samples/vboxwrapper/Makefile
+++ boinc/samples/vboxwrapper/Makefile
@@ -27,25 +27,26 @@ distclean:
 	/bin/rm -f $(PROGS) *.o libstdc++.a
 
 floppyio.o: floppyio.cpp
-	g++ $(CXXFLAGS) -c floppyio.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c floppyio.cpp
 
 vbox_common.o: vbox_common.cpp
-	g++ $(CXXFLAGS) -c vbox_common.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vbox_common.cpp
 
 vbox_vboxmanage.o: vbox_vboxmanage.cpp
-	g++ $(CXXFLAGS) -c vbox_vboxmanage.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vbox_vboxmanage.cpp
 
 vboxcheckpoint.o: vboxcheckpoint.cpp
-	g++ $(CXXFLAGS) -c vboxcheckpoint.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vboxcheckpoint.cpp
 
 vboxjob.o: vboxjob.cpp
-	g++ $(CXXFLAGS) -c vboxjob.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vboxjob.cpp
 
 vboxlogging.o: vboxlogging.cpp
-	g++ $(CXXFLAGS) -c vboxlogging.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vboxlogging.cpp
 
 vboxwrapper.o: vboxwrapper.cpp
-	g++ $(CXXFLAGS) -c vboxwrapper.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c vboxwrapper.cpp
 
 vboxwrapper: vboxwrapper.o vbox_common.o vbox_vboxmanage.o vboxcheckpoint.o vboxjob.o vboxlogging.o floppyio.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a
-	g++ $(CXXFLAGS) -o vboxwrapper vboxwrapper.o vbox_common.o vbox_vboxmanage.o vboxcheckpoint.o vboxjob.o vboxlogging.o floppyio.o libstdc++.a -pthread -lboinc_api -lboinc
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -static -pthread -o vboxwrapper vboxwrapper.o vbox_common.o vbox_vboxmanage.o vboxcheckpoint.o vboxjob.o vboxlogging.o floppyio.o \
+	libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
Index: boinc/samples/worker/Makefile
===================================================================
--- boinc.orig/samples/worker/Makefile
+++ boinc/samples/worker/Makefile
@@ -10,4 +10,4 @@ distclean:
 	rm -f $(PROGS) *.o
 
 worker: worker.o
-	g++ $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o worker worker.o
+	$(CXX) $(CPPFLAGS) -static -o worker worker.o
Index: boinc/samples/wrapper/Makefile
===================================================================
--- boinc.orig/samples/wrapper/Makefile
+++ boinc/samples/wrapper/Makefile
@@ -37,5 +37,5 @@ REGEXP_OBJS = \
 	regexp_memory.o \
 	regexp_report.o
 
-wrapper: wrapper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(REGEXP_OBJS)
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) -o wrapper wrapper.o $(REGEXP_OBJS) libstdc++.a $(PTHREAD) -lboinc_api -lboinc -lboinc_zip $(LDFLAGS)
+wrapper: wrapper.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_ZIP_DIR)/libboinc_zip.a $(REGEXP_OBJS)
+	$(CXX) $(CPPFLAGS) $(PTHREAD) -static -o wrapper wrapper.o $(REGEXP_OBJS) libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_ZIP_DIR)/libboinc_zip.a $(BOINC_LIB_DIR)/libboinc.a
Index: boinc/samples/wrappture/Makefile
===================================================================
--- boinc.orig/samples/wrappture/Makefile
+++ boinc/samples/wrappture/Makefile
@@ -29,10 +29,10 @@ distclean:
 	/bin/rm -f $(PROGS) *.o libstdc++.a
 
 wrappture.o: wrappture.cpp
-	g++ $(CXXFLAGS) -c -o wrappture.o wrappture.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c -o wrappture.o wrappture.cpp
 
 wrappture_example: wrappture.o wrappture_example.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_api.a $(RAPPTURE_DIR)/librappture.a
-	g++ $(CXXFLAGS) -o wrappture_example wrappture_example.o wrappture.o $(RAPPTURE_DIR)/librappture.a libstdc++.a -pthread -lboinc_api -lboinc -lexpat -lz
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -static -pthread -o wrappture_example wrappture_example.o wrappture.o $(RAPPTURE_DIR)/librappture.a libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
 
 fermi: fermi.cpp
-	g++ $(CSSFLAGS) -o fermi fermi.cpp
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -static -o fermi fermi.cpp
Index: boinc/samples/example_app/Makefile
===================================================================
--- boinc.orig/samples/example_app/Makefile
+++ boinc/samples/example_app/Makefile
@@ -61,17 +61,17 @@ install: uc2
 # because otherwise you might get a version in /usr/lib etc.
 
 uc2: uc2.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2 uc2.o libstdc++.a -pthread \
+	$(CXX) $(CPPFLAGS) -static -pthread -o uc2 uc2.o libstdc++.a \
 	$(BOINC_API_DIR)/libboinc_api.a \
 	$(BOINC_LIB_DIR)/libboinc.a
 
 ucn: ucn.o libstdc++.a $(BOINC_API_DIR)/libboinc_api.a $(BOINC_LIB_DIR)/libboinc.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o ucn ucn.o libstdc++.a -pthread \
+	$(CXX) $(CPPFLAGS) -static -pthread -o ucn ucn.o libstdc++.a \
 	$(BOINC_API_DIR)/libboinc_api.a \
 	$(BOINC_LIB_DIR)/libboinc.a
 
 uc2_graphics: uc2_graphics.o ttfont.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_graphics2.a
-	$(CXX) $(CXXFLAGS) $(CPPFLAGS) $(LDFLAGS) -o uc2_graphics uc2_graphics.o ttfont.o libstdc++.a -pthread \
+	$(CXX) $(CPPFLAGS) -static -pthread -o uc2_graphics uc2_graphics.o ttfont.o libstdc++.a \
 	$(BOINC_API_DIR)/libboinc_graphics2.a \
 	$(BOINC_API_DIR)/libboinc_api.a \
 	$(BOINC_LIB_DIR)/libboinc.a \
@@ -79,7 +79,7 @@ uc2_graphics: uc2_graphics.o ttfont.o li
     -lftgl -lGL -lX11 -lXmu -lm
 
 slide_show: slide_show.o libstdc++.a $(BOINC_LIB_DIR)/libboinc.a $(BOINC_API_DIR)/libboinc_graphics2.a
-	$(CXX) $(CXXFLAGS) -o slide_show slide_show.o libstdc++.a -pthread \
+	$(CXX) -static -pthread -o slide_show slide_show.o libstdc++.a \
 	$(BOINC_API_DIR)/libboinc_graphics2.a \
 	$(BOINC_API_DIR)/libboinc_api.a \
 	$(BOINC_LIB_DIR)/libboinc.a \
Index: boinc/api/Makefile.am
===================================================================
--- boinc.orig/api/Makefile.am
+++ boinc/api/Makefile.am
@@ -45,6 +45,10 @@ lib_LTLIBRARIES = libboinc_api.la
 libboinc_api_la_SOURCES = $(api_files)
 libboinc_api_la_LDFLAGS = -version-number $(LIBBOINC_VERSION) -lpthread
 libboinc_api_la_LIBADD = -L../lib/.libs $(LIBBOINC)
+libboinc_api_la_AR = gcc-ar
+libboinc_api_la_RANLIB = gcc-ranlib
+libboinc_api_a_AR = gcc-ar
+libboinc_api_a_RANLIB = gcc-ranlib
 
 if BUILD_GRAPHICS_API
 lib_LTLIBRARIES += libboinc_graphics2.la
@@ -52,12 +56,20 @@ libboinc_graphics2_la_SOURCES = $(graphi
 libboinc_graphics2_la_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/lib -I$(top_srcdir)/samples/image_libs
 libboinc_graphics2_la_LDFLAGS = -version-number $(LIBBOINC_VERSION) -ljpeg
 libboinc_graphics2_la_LIBADD = -L../lib/.libs $(APPLIBS)
+libboinc_graphics2_la_AR = gcc-ar
+libboinc_graphics2_la_RANLIB = gcc-ranlib
+libboinc_graphics2_a_AR = gcc-ar
+libboinc_graphics2_a_RANLIB = gcc-ranlib
 endif #BUILD_GRAPHICS_API
 
 lib_LTLIBRARIES += libboinc_opencl.la
 libboinc_opencl_la_SOURCES = $(opencl_files)
 libboinc_opencl_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)
 libboinc_opencl_la_LIBADD = -L../lib/.libs $(APPLIBS)
+libboinc_opencl_la_AR = gcc-ar
+libboinc_opencl_la_RANLIB = gcc-ranlib
+libboinc_opencl_a_AR = gcc-ar
+libboinc_opencl_a_RANLIB = gcc-ranlib
 
 if INSTALL_HEADERS
 ## install only headers that are meant for exporting the API !!
Index: boinc/zip/Makefile.am
===================================================================
--- boinc.orig/zip/Makefile.am
+++ boinc/zip/Makefile.am
@@ -63,6 +63,10 @@ lib_LTLIBRARIES = libboinc_zip.la
 libboinc_zip_la_SOURCES = $(libboinc_zip_sources)
 libboinc_zip_la_LDFLAGS = -version-number $(LIBBOINC_VERSION)
 libboinc_zip_la_LIBADD = $(LIBBOINC)
+libboinc_zip_la_AR = gcc-ar
+libboinc_zip_la_RANLIB = gcc-ranlib
+libboinc_zip_a_AR = gcc-ar
+libboinc_zip_a_RANLIB = gcc-ranlib
 
 # Some OSs may not prefix libraries with lib.
 # For example OS2
Index: boinc/configure.ac
===================================================================
--- boinc.orig/configure.ac
+++ boinc/configure.ac
@@ -69,6 +69,12 @@ SAH_LINKS
 AC_LANG_PUSH(C)
 AM_PROG_CC_C_O
 
+dnl ------
+dnl  essential for LTO and static applications in apps and samples
+AC_PROG_RANLIB
+AM_PROG_AR
+dnl ------
+
 m4_divert_once([HELP_ENABLE],
   AS_HELP_STRING([BOINC Default enable values], [--enable-server --enable-client --enable-libraries --enable-manager: builds server, client, and libraries]))
 
